mwim.el

```diff
@@ -124,38 +124,63 @@ to the end of line."
           (goto-char eoc)))))
   (skip-chars-backward " \t"))
 
+(defun mwim-forward-line (arg)
+  "Move ARG - 1 lines forward (backward if ARG is negative).
+If ARG is nil or 1 do nothing."
+  (interactive "^p")
+  (setq arg (or arg 1))
+  (when (/= arg 1)
+    (let ((line-move-visual nil))
+      (forward-line (1- arg)))))
+
 ;;;###autoload
-(defun mwim-beginning-of-code-or-line ()
+(defun mwim-beginning-of-code-or-line (arg)
   "Move point to the beginning of code.
-If the point is already there, move to the beginning of line."
-  (interactive "^")
+If the point is already there, move to the beginning of line.
+
+If ARG is not nil or 1, move forward ARG - 1 lines first. If
+point reaches the beginning or end of the buffer, stop there."
+  (interactive "^p")
+  (call-interactively #'mwim-forward-line)
```

Comment:
This is not needed, just: ``(mwim-forward-line arg)``.

In general, `call-interactively` is needed very rarely, and (IMHO) most
of the times it is a sign of a bad code :-)